luigi/parameter.py

```diff
@@ -331,9 +337,35 @@ class MonthParameter(DateParameter):
 
     date_format = '%Y-%m'
 
-    @staticmethod
-    def next_in_enumeration(_value):
-        return None
+    def _add_months(self, date, months):
+        """
+        Add ``months`` months to ``date``.
+
+        Unfortunately we can't use timedeltas to add months because timedelta counts in days
+        and there's no foolproof way to add N months in days without counting the number of
+        days per month.
+        """
+        year = date.year + (months // 12)
```

Comment:
Hmm I don't think that works with something like 12/2015 + 2 == 2/2016 ?

Separately, the month < 1 logic is for correctness in the value < self.start case (which seems like a bad idea, but I'd rather it not fail :D).